cmd/proxy-server/main.go

```diff
@@ -540,3 +563,152 @@ func rolesHandler(log *logrus.Entry, opaHost string) http.Handler {
 		defer res.Body.Close()
 	})
 }
+
+func volumesHandler(roleServ *roleClientService, tm token.Manager, log *logrus.Entry) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		var sysID, sysType, storPool, tenant string
+		var volumeMap = make(map[string]string)
+		var volumeList []string
+		var resp *pb.RoleListResponse
+		keyTenantRevoked := "tenant:revoked"
+
+		// Verify token
+		log.Info("Verifying token!")
+
+		authz := r.Header.Get("Authorization")
+		parts := strings.Split(authz, " ")
+		if len(parts) != 2 {
+			log.Println("invalid authz header")
+			return
```

Comment:
Can we write a JSON error response here? If the token is not sent properly, there would be no error on the client.

Reply:
fixed